antarest/study/service.py

```diff
@@ -1287,7 +1287,7 @@ def delete_study(self, uuid: str, children: bool) -> None:
             _ = study.workspace
             study_info = study.to_enhanced_json_summary()
 
-        if self.storage_service.variant_study_service.has_children(study):
```

Comment:
hrer we don't need to cast: actually the `has_children` method should expect a `Study` type, because any type of study can have children, not only variants;